addons/purchase/tests/test_purchase_lead_time.py

```diff
@@ -0,0 +1,165 @@
+        self.StockPicking = self.env['stock.picking']
+
+        self.company = self.env.ref('base.main_company')
+        self.partner = self.env.ref('base.res_partner_18')
+        self.uom_unit = self.env.ref('product.product_uom_unit')
+        self.warehouse = self.env.ref('stock.warehouse0')
+
+        route_buy = self.warehouse.buy_pull_id.route_id.id
+        route_mto = self.warehouse.mto_pull_id.route_id.id
+
+        # Create product A with Delivery Lead Time
+        self.productA = self.Product.create({
+            'name': 'Product A',
+            'type': 'product',
+            'uom_id': self.uom_unit.id,
+            'uom_po_id': self.uom_unit.id,
+            'route_ids': [(6, 0, [route_buy, route_mto])],
+            'seller_ids': [(0, 0, {'name': self.partner.id, 'delay': 5})]})
+
+        # Create product B with Delivery Lead Time
+        self.productB = self.Product.create({
+            'name': 'Product B',
+            'type': 'product',
+            'uom_id': self.uom_unit.id,
+            'uom_po_id': self.uom_unit.id,
+            'route_ids': [(6, 0, [route_buy, route_mto])],
+            'seller_ids': [(0, 0, {'name': self.partner.id, 'delay': 2})]})
+
+    def _create_procurement(self, product, productqty):
+        return self.env['make.procurement'].create({
+            'product_id': product,
+            'qty': productqty,
+            'uom_id': self.uom_unit.id,
+            'warehouse_id': self.warehouse.id,
+            'date_planned': fields.Datetime.to_string(fields.datetime.now() + timedelta(days=10))})  # 10 days added to current date of procurement to get future schedule date and order date of purchase order.
+
+    def test_00_product_company_level_delays(self):
+        """ To check dates, set product's Delivery Lead Time and company's Purchase Lead Time."""
+
+        # company with Purchase Lead Time
+        self.company.write({'po_lead': 3.00})
+
+        # Create procurement order of product A
+        procurement_productA = self._create_procurement(self.productA.id, 15.00)
+        procurement_act_dict = procurement_productA.make_procurement()
+        procurement = self.ProcurementOrder.browse(procurement_act_dict.get("res_id"))
+        self.assertEqual(procurement.state, 'running', 'Procurement should be in running state.')
+
+        # Confirm purchase order
+        procurement.purchase_id.button_confirm()
+
+        # Check scheduled date and order date of purchase order
+        order_date = fields.Datetime.from_string(procurement.date_planned) - timedelta(days=self.company.po_lead) - timedelta(days=self.productA.seller_ids.delay)
+        po_order_date = fields.Datetime.to_string(order_date)
+        schedule_date = order_date + timedelta(days=self.productA.seller_ids.delay)
+        po_schedule_date = fields.Datetime.to_string(schedule_date)
+        self.assertEqual(procurement.purchase_id.date_order, po_order_date, 'Wrong order date of purchase order.')
```

Comment:
Example of unnecessary verbose text. The string does not give anything meaningfull.